src/software/geom/algorithms/end_in_obstacle_sample.cpp

```diff
@@ -0,0 +1,51 @@
+#include <random>
+#include "software/geom/algorithms/end_in_obstacle_sample.h"
+#include "software/geom/point.h"
+#include "software/ai/navigator/obstacle/obstacle.hpp"
+
+
+Point endInObstacleSample(const std::vector<ObstaclePtr> obstacles, const Point &destination, const Rectangle &navigable_area, double multiplier, double range)
```

Comment:
nit: lets generalize the names a little bit so it's not specific to just end position. Since this could technically work with any point. (same with the function name)
```suggestion
Point endInObstacleSample(const std::vector<ObstaclePtr>& obstacles, const Point &point, const Rectangle &navigable_area, double multiplier, double range)
```